cloud/scope/machine_test.go

```diff
@@ -147,6 +147,17 @@ func TestSetVPCProviderID(t *testing.T) {
 		err := scope.SetProviderID(core.StringPtr(providerID))
 		g.Expect(err).To(BeNil())
 	})
+
+	t.Run("Set Provider ID returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		scope := setupMachineScope(clusterName, machineName, mock.NewMockVpc(gomock.NewController(t)))
+		options.ProviderIDFormat = string("v2")
+		utils.GetAccountIDFunc = func() (string, error) {
+			return "", errors.New("error getting accountID") // Return dummy error
+		}
+		err := scope.SetProviderID(core.StringPtr(providerID))
```

Comment:
We are slowly migrating to use ptr.To() instead of core.StringPtr() can we do that here as well?